### PR TITLE
Add reset windows password

### PR DESCRIPTION
**Pull request by victorursan**
- added reset windows password
- WIP set metadata sync ... to decide the outputs

### CODE REVIEW

File: cs-google/pom.xml [hunk 1]
```diff
@@ -56,6 +56,9 @@
     <properties>
         <specs2.version>3.9.5</specs2.version>
         <commons-io.version>2.5</commons-io.version>
+        <spray-json.version>1.3.3</spray-json.version>
+        <cs-commons.version>0.0.3</cs-commons.version>
```
Comment (ionutmarisca) [inline cs-google/pom.xml]: Why is this here?

File: cs-google/src/main/scala/io/cloudslang/content/google/actions/compute/compute_engine/instances/InstancesResetWindowsPassword.scala [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    *                           Default: "30"
+    * @param pollingIntervalInp Optional - The time, in seconds, to wait before a new request that verifies if the operation
+    *                           finished is executed, if the sync input is set to "true".
+    *                           Valid values: Any positive number including 0.
+    *                           Default: "1"
+    * @param proxyPortInp       Optional - Proxy server port used to access the provider services.
+    *                           Default: "8080"
+    * @param proxyUsername      Optional - Proxy server user name.
+    * @param proxyPasswordInp   Optional - Proxy server password associated with the <proxyUsername> input value.
+    * @return a map containing a Instance resource as returnResult
```
Comment (ionutmarisca) [inline cs-google/src/main/scala/io/cloudslang/content/google/actions/compute/compute_engine/instances/InstancesResetWindowsPassword.scala]: I think this should be the password, not the instance

File: cs-google/src/main/scala/io/cloudslang/content/google/actions/compute/compute_engine/instances/InstancesSetMetadata.scala [hunk 3]
```diff
@@ -90,28 +108,43 @@ class InstancesSetMetadata {
...
       val httpTransport = HttpTransportUtils.getNetHttpTransport(proxyHostOpt, toInteger(proxyPortStr), proxyUsernameOpt, proxyPassword)
       val jsonFactory = JsonFactoryUtils.getDefaultJacksonFactory
       val credential = GoogleAuth.fromAccessToken(accessToken)
 
       val items: List[Items] = InstanceController.createMetadataItems(itemsKeysList, itemsValuesList, itemsDelimiter)
 
-      val result = InstanceService.setMetadata(httpTransport, jsonFactory, credential, projectId, zone, instanceName, items)
-      val resultString = if (toBoolean(prettyPrint)) result.toPrettyString else result.toString
+      val result = InstanceService.setMetadata(httpTransport, jsonFactory, credential, projectId, zone, instanceName, items, sync, timeout, pollingInterval) //TODO
+      val resultString = if (prettyPrint) result.toPrettyString else result.toString
```
Comment (ionutmarisca) [inline cs-google/src/main/scala/io/cloudslang/content/google/actions/compute/compute_engine/instances/InstancesSetMetadata.scala]: Use toPretty method

File: cs-google/src/main/scala/io/cloudslang/content/google/actions/compute/compute_engine/instances/InstancesAttachDisk.scala [hunk 4]
```diff
@@ -9,7 +9,8 @@ import io.cloudslang.content.constants.OutputNames.{EXCEPTION, RETURN_CODE, RETU
 import io.cloudslang.content.constants.{ResponseNames, ReturnCodes}
 import io.cloudslang.content.google.services.compute.compute_engine.disks.DiskController
 import io.cloudslang.content.google.services.compute.compute_engine.instances.InstanceService
-import io.cloudslang.content.google.utils.Constants.{NEW_LINE, TIMEOUT_EXCEPTION}
+import io.cloudslang.content.google.utils.Constants
```
Comment (robert-sandor) [inline cs-google/src/main/scala/io/cloudslang/content/google/actions/compute/compute_engine/instances/InstancesAttachDisk.scala]: why do you import Constants, when you already have some imported below ?

File: cs-google/src/main/scala/io/cloudslang/content/google/actions/compute/compute_engine/instances/InstancesGet.scala [hunk 5]
```diff
@@ -2,22 +2,29 @@ package io.cloudslang.content.google.actions.compute.compute_engine.instances
...
 import java.util
 
+import com.google.api.services.compute.model.NetworkInterface
 import com.hp.oo.sdk.content.annotations.{Action, Output, Param, Response}
 import com.hp.oo.sdk.content.plugin.ActionMetadata.{MatchType, ResponseType}
 import io.cloudslang.content.constants.OutputNames.{EXCEPTION, RETURN_CODE, RETURN_RESULT}
 import io.cloudslang.content.constants.{ResponseNames, ReturnCodes}
 import io.cloudslang.content.google.services.compute.compute_engine.instances.InstanceService
-import io.cloudslang.content.google.utils.Constants.NEW_LINE
+import io.cloudslang.content.google.utils.Constants
```
Comment (robert-sandor) [inline cs-google/src/main/scala/io/cloudslang/content/google/actions/compute/compute_engine/instances/InstancesGet.scala]: same as above

File: cs-google/src/main/scala/io/cloudslang/content/google/actions/compute/compute_engine/instances/InstancesGet.scala [hunk 6]
```diff
@@ -44,12 +51,19 @@ class InstancesGet {
     * @param prettyPrintInp   Optional - Whether to format (pretty print) the resulting json.
     *                         Valid values: "true", "false"
     *                         Default: "true"
-    * @return a map containing a Instance resource as returnResult
+    * @return a map containing a Instance resource as returnResult, the
```
Comment (robert-sandor) [inline cs-google/src/main/scala/io/cloudslang/content/google/actions/compute/compute_engine/instances/InstancesGet.scala]: i think this should be continued 😆

File: cs-google/src/main/scala/io/cloudslang/content/google/actions/compute/compute_engine/instances/InstancesGet.scala [hunk 7]
```diff
@@ -89,9 +103,22 @@ class InstancesGet {
       val credential = GoogleAuth.fromAccessToken(accessToken)
 
       val instance = InstanceService.get(httpTransport, jsonFactory, credential, projectId, zone, instanceName)
-      val resultString = if (prettyPrint) instance.toPrettyString else instance.toString
+      val metadata =  Option(instance.getMetadata.getItems).getOrElse(List().asJava)
+      val tags =  Option(instance.getTags.getItems).getOrElse(List().asJava)
+      val networkInterfaces = Option(instance.getNetworkInterfaces).getOrElse(List[NetworkInterface]().asJava)
```
Comment (robert-sandor) [inline cs-google/src/main/scala/io/cloudslang/content/google/actions/compute/compute_engine/instances/InstancesGet.scala]: is it necessary to specify the type of List, `NetworkInterface` here ?

**Comment by robert-sandor** [comment]
grate job 👍  🧀
